### PR TITLE
(BOLT-333) Add canary and aggregate modules to /modules

### PR DESCRIPTION
**Pull request by adreyer**
Bolt now appends it's internal modules directory to the modulepath which
includes the canary module.

### CODE REVIEW

**Comment by MikaelSmith** [comment]
I've fixed up what's here. I plan to add an aggregate plan as well.

**Comment by MikaelSmith** [comment]
I've updated this to include the aggregate plans. I believe it's ready for further review.

File: modules/canary/plans/init.pp [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  TargetSpec $nodes,
+  Hash[String, Data] $params = {},
+  Integer $canary_size = 1
+) {
+
+  # Validation
+  $type_count = [$task, $command, $script].reduce(0) |$acc, $v| {
+    if($v) {
+      $acc + 1
+    } else { $acc } }
```
Comment (mruzicka) [inline modules/canary/plans/init.pp]: again bad code style

File: modules/canary/plans/init.pp [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  if($type_count == 0){
```
Comment (mruzicka) [inline modules/canary/plans/init.pp]: missing space before opening curly brace

File: modules/canary/plans/init.pp [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    fail_plan("Must specify a command, script, or task to run", 'canary/invalid-params')
+  }
+
+  if($type_count > 1) {
+    fail_plan("Must specify only one command, script, or task to run", 'canary/invalid-params')
+  }
+
+  [$canaries, $rest] = canary::random_split(get_targets($nodes), $canary_size)
+  $catch_params = $params + { '_catch_errors' => true}
```
Comment (mruzicka) [inline modules/canary/plans/init.pp]: missing space before closing curly brace

**Comment by MikaelSmith** [comment]
Fixed up.

**Pull request merged**